samples/cpp/autofocus.cpp

```diff
@@ -244,60 +244,36 @@ static void showHelp(const char * pName, bool welcomeMsg)
 
 static bool parseArguments(int argc, char ** argv)
 {
-    int index;
-    GlobalArgs.deviceName = "Nikon";
-    GlobalArgs.output = NULL;
-    GlobalArgs.fps = DEFAULT_OUTPUT_FPS;
-    GlobalArgs.minimumFocusStep = 0;
+    cv::CommandLineParser parser(argc, argv, "{h help||}{o||}{f||}{m||}{d|0|}{v||}{@device|Nikon|}");
+    if (parser.has("help"))
+        return false;
     GlobalArgs.breakLimit = DEFAULT_BREAK_LIMIT;
-    GlobalArgs.measure = false;
-    GlobalArgs.verbose = false;
-
-    for (index = 1; index < argc; index++)
+    if (parser.has("o"))
+        GlobalArgs.output = parser.get<string>("o");
+    else
+        GlobalArgs.output = "";
+    if (parser.has("f"))
+        GlobalArgs.fps = parser.get<int>("f");
+    else
+        GlobalArgs.fps = DEFAULT_OUTPUT_FPS;
+    GlobalArgs.measure = parser.has("m");
+    GlobalArgs.verbose = parser.has("v");
+    GlobalArgs.minimumFocusStep = parser.get<int>("d");
+    GlobalArgs.deviceName = parser.get<string>("@device");
+    if (!parser.check())
     {
-        const char * arg = argv[index];
-        if (strcmp(arg, "-h") == 0)
-        {
-            return false;
-        }
-        else if (strcmp(arg, "-o") == 0)
-        {
-            GlobalArgs.output = argv[++index];
-        }
-        else if (strcmp(arg, "-f") == 0)
-        {
-            if (sscanf(argv[++index], "%u", &GlobalArgs.fps) != 1
-                    || GlobalArgs.fps <= 0)
-            {
-                cerr << "Invalid fps argument." << endl;
-                return false;
-            }
-        }
-        else if (strcmp(arg, "-m") == 0)
-        {
-            GlobalArgs.measure = true;
-        }
-        else if (strcmp(arg, "-v") == 0)
-        {
-            GlobalArgs.verbose = true;
-        }
-        else if (strcmp(arg, "-d") == 0)
-        {
-            if (sscanf(argv[++index], "%u", &GlobalArgs.minimumFocusStep) != 1
-                    || GlobalArgs.minimumFocusStep <= 0)
-            {
-                cerr << "Invalid minimum focus step argument." << endl;
-                return false;
-            }
-        }
-        else if (arg[0] != '-')
-        {
-            GlobalArgs.deviceName = arg;
-        }
-        else
-        {
-            cerr << "Unknown option " << arg << endl;
-        }
+        parser.printErrors();
+        return false;
+    }
+    if (GlobalArgs.fps <= 0)
+    {
+        cerr << "Invalid fps argument." << endl;
+        return false;
+    }
+    if (GlobalArgs.minimumFocusStep <= 0)
```

Comment:
Actually, _minimumFocusStep_ and _fps_ parameters can be set to 0. 
Previous logic was: `IF "-d" is provided THEN it should be > 0 ELSE it is 0` ; 
your logic is: `"-d" should be > 0`

I think it is safe to replace this expression to: `if (GlobalArgs.minimumFocusStep < 0)`. Same for `GlobalArgs.fps`